FWCore/Services/src/SiteLocalConfigService.cc

```diff
@@ -26,9 +26,36 @@ using namespace xercesc;
 //<<<<<< MEMBER FUNCTION DEFINITIONS                                    >>>>>>
 
 namespace {
+
+  template< class CharType >
+  class ZStr      // Zero-terminated string.
+  {
+  public:
+    ZStr(CharType const* str, void (*deleter)(CharType*))
```

Comment:
Given that you know what the deleter should be, I'd suggest doing
```C++
ZStr(CharType const* str):
   m_array(const_cast<CharType*>(str), &dispose) {}
```
Where you'd have to move the declaration of `dispose` to before `ZStr`.